Changed FindFlyCapture2.cmake so it works in Windows as well

VileLasagna opened the pull request:
The initial form of this saved me quite some pain. Thought I'd give back my own improvements on it.

VileLasagna committed:
Changed FindFlyCapture2.cmake so it works in Windows as well

kartikmohta commented:
@VileLasagna You're using the ROS driver in Windows?

VileLasagna commented:
I'm not really using your drivers. I just needed to use Flycap in a different project and found your CMake script. So this is basically all I know from this repo

versatran01 commented:
@kartikmohta You can merge this if you think this is ok.

kartikmohta commented on cmake/FindFlyCapture2.cmake:
```diff
@@ -71,11 +79,31 @@ list(APPEND FlyCapture2_CHECK_INCLUDE_DIRS
 list(APPEND FlyCapture2_CHECK_LIBRARY_DIRS
     ${FLYCAPTURE_DIR}/flycapture/lib
     )
+if(MSVC)
```
Shouldn't this be `if(WIN32)`?

kartikmohta commented on cmake/FindFlyCapture2.cmake:
```diff
@@ -84,67 +112,162 @@ set(FlyCapture2_FOUND TRUE)
-    FlyCapture2_REPORT_NOT_FOUND("Caller defined FlyCapture2_INCLUDE_DIR: "
-        ${FlyCapture2_INCLUDE_DIR}
-        " does not contain flycapture/FlyCapture22.h header.")
+if(MSVC)
```
`if(WIN32)` here?

kartikmohta commented on cmake/FindFlyCapture2.cmake:
```diff
@@ -84,67 +112,162 @@ set(FlyCapture2_FOUND TRUE)
     )
+else()
+
+    if(MSVC)
```
I think the `if(MSVC)` is only needed for the following block for setting `vTail` but not needed for the later `find_library` and `find_file`.

kartikmohta commented on cmake/FindFlyCapture2.cmake:
```diff
@@ -84,67 +112,162 @@ set(FlyCapture2_FOUND TRUE)
 if(FlyCapture2_FOUND)
     set(FlyCapture2_INCLUDE_DIRS ${FlyCapture2_INCLUDE_DIR})
-    file(GLOB FlyCapture2_LIBRARIES ${FlyCapture2_LIBRARY})
+    if(MSVC)
```
`if(WIN32)`?

kartikmohta reviewed (commented):
@VileLasagna Have a few comments about the use of `if(MSVC)` vs `if(WIN32)`. Let me know if you agree with the proposed changes.

VileLasagna commented:
Hey

The reason I've hidden these behind MSVC checks is because of CYGWIN and MINGW. They're both windows platforms but I'm honestly not sure how they'd link to these, if you were using one of those environments (something like clang-cl still counts as MSVC because of vcvarsall.bat).

My home machines are running only Linux and at work it's all VC so I don't have the environment properly set up to quickly test those so it's more me not remembering how this goes once you go down that path. A quick googling says you're probably OK if you're linking with C idiom so... I can check Flycap tomorrow and if it's C idiom default to the vc140 version, toss a warning and hope for the best?

kartikmohta commented:
The places I've marked are not dependent on the compiler being used (for example, when finding the location of header files or lib files) so hopefully they should be fine with MSVC or MINGW. Cygwin most likely won't work at all for this.

VileLasagna commented:
I still think it shouldn't link but if you think it's a better option I can have it point to the newest library, throw a warning and hope for the  best?

kartikmohta commented:
I'm not sure either whether it will link, but for now I suppose you can point to the newest one and we'll hope that somebody comes along, tests and reports a bug if it doesn't work 🙂

VileLasagna committed:
Further tweaked FindFlyCapture2.cmake for Windows

Instead of giving you nothing if you're not using MSVC (MINGW or CYGWIN)
the script will instead give you the latest version (VC14) and throw a
warning your way, also urging you to consider using the C idiom version
of the libraries

VileLasagna commented:
I've implemented the changes we talked about. 
As part of the warning message I've suggested the user look into the C Idiom libraries Point Grey provides as part of their FlyCapture package. Since they're different libraries, however, I haven't adapted the script to actually give you those though, but they should be a more hopeful option for people trying to link FlyCap while building with CYGWIN or MINGW

kartikmohta commented:
Thanks @VileLasagna
